src/mono/wasm/runtime/invoke-cs.ts

```diff
@@ -242,14 +319,22 @@ function bind_fn(closure: BindingClosure) {
                     marshaler(args, js_arg);
                 }
             }
+            let js_result = undefined;
+            if (is_async && res_converter) {
+                // pre-allocate the promise
+                js_result = res_converter(args);
```

Comment:
What does it mean if it's async but there's no res_converter? shouldn't there still be a pre-allocated promise? the is_async below doesn't check for res_converter

Reply:
I was just lazy. I will uses `!` to convince type system at compile time.